### ISSUE TITLE
Purpose of Supercompressed Global Data block

### ISSUE THREAD

**Issue by dewilkinson** [KhronosGroup/KTX-Specification#34]
Looking for clarification on this one.

IIUC the SupercompressedGlobalData array block is designed to hold codebooks for the Huffman encoded portions of the compressed data stream 

My question: the codebook should be encapsulated within the encoder. Only the decoder, not the app can use this data. It is also already contained in the bitstream - so why are we copying it out into the body of the file format where it is of no practical use? The codebook will be extracted by the decoder from the bitstream directly and is the only place it would be used.

Am I missing something that would make it necessary to pull codebook data up to the file format level?

What I *do* see a use for is simply a global block of opaque binary data that can be optionally used and populated with binary data meaningful to the target app that may be useful - but certainly not limited to codebooks or even compression

In summary, I find Supercompressed Global Data block redundant , but I am in favor of a general purpose opaque ‘Global Data’ block. 
So Effectively , the only change I’d propose would be the naming , having that binary block there is still useful...

**Comment by MarkCallow** [comment]
Crunch CRN has a code book common to all mip levels. Before we can inflate any mip level we must have the global code book. There are 2 ways to accomplish this: put it at the file format level, or specify that the first mip level transmitted must contain the code book. The first seemed more reliable and easier to specify when the ordering of the mip levels can change. But it is not a strong preference.

Apps adding their own global data is already supported. It's called metadata.

**Comment by MarkCallow** [comment]
Another reason for having the global code book at the file format level is it is less likely to be missed by those using, e.g., HTTP, to read ranges from the file. If it is hidden in the first mip level, a user may directly read a later miplevel then discover they can't inflate it.

**Comment by dewilkinson** [comment]
Understood. I would like to discuss this one some more, but it is not a blocking issue - the data block should be supported, its just clarification on the various things we'd need to use it for

**Comment by dewilkinson** [comment]
On the 12/21 call we reached consensus that the supercompressed global data should not be made available at the file level, given that the data should be abstracted in the decoder. The codebook binary block will instead be relocated to be the first data block immediately prior to the highest level mip level in the data stream. 


This issue will be closed, and a new one opened to track the work item

**Comment by MarkCallow** [comment]
This is not my understanding. We agreed to move the block but it would still be visible at the file level due to wanting random access to the mip levels. Let's continue further discussion on issue #39 you opened to track to work.
